deps/build.jl

```diff
@@ -1,125 +1,60 @@
+    Linux(:aarch64, :musl) => ("$bin_prefix/ImageMagick.v6.9.10-4.aarch64-linux-musl.tar.gz", "ccdc95d46c5e0412d754138f7508f59cda0b93ea31c7edcf300dc37f9aaf172c"),
+    Linux(:armv7l, :glibc, :eabihf) => ("$bin_prefix/ImageMagick.v6.9.10-4.arm-linux-gnueabihf.tar.gz", "6373eab42d3866ae7618ab2d6ed459265f8eda71fbd06466c25c4a36f938509c"),
+    Linux(:armv7l, :musl, :eabihf) => ("$bin_prefix/ImageMagick.v6.9.10-4.arm-linux-musleabihf.tar.gz", "249b64e55165a2a9fdc5ed805d8682ed8bebb1d54d0b2e2af1860002e73a8d10"),
+    Linux(:i686, :glibc) => ("$bin_prefix/ImageMagick.v6.9.10-4.i686-linux-gnu.tar.gz", "634f1c8eac84d34ecd2def7cec19835fe5d1d696157acffccd751b9e6214d889"),
+    Linux(:i686, :musl) => ("$bin_prefix/ImageMagick.v6.9.10-4.i686-linux-musl.tar.gz", "acf52b51eef9c839e5e4b8e3a8fa77c5ded1be4be0628f85f55cdad90bb8771d"),
+    Windows(:i686) => ("$bin_prefix/ImageMagick.v6.9.10-4.i686-w64-mingw32.tar.gz", "06d7bb43cab8b6d760290040f6382160409d0187fe7a71ece32bc4fb6333995d"),
+    Linux(:powerpc64le, :glibc) => ("$bin_prefix/ImageMagick.v6.9.10-4.powerpc64le-linux-gnu.tar.gz", "40399f265034d7aa0c522057af001e437da50b50aa23c739c4fd38b6185da5f9"),
+    MacOS(:x86_64) => ("$bin_prefix/ImageMagick.v6.9.10-4.x86_64-apple-darwin14.tar.gz", "8b0461ded10b9ce04baae7271feb610f25673abc43d555b9a6651a4c127bbf41"),
+    Linux(:x86_64, :glibc) => ("$bin_prefix/ImageMagick.v6.9.10-4.x86_64-linux-gnu.tar.gz", "06bae488abadea7e325cf75766cd67574ec3ccd54ae064675efb5b2a89f54eac"),
+    Linux(:x86_64, :musl) => ("$bin_prefix/ImageMagick.v6.9.10-4.x86_64-linux-musl.tar.gz", "01ba7e01ae965f75b752206d994c52946d24241c5a8c0668e6d9407de6239a77"),
+    FreeBSD(:x86_64) => ("$bin_prefix/ImageMagick.v6.9.10-4.x86_64-unknown-freebsd11.1.tar.gz", "c33ad0f46ede198bd16459d20d741ec65153e05b9ddd637907567c574a157ed8"),
+    Windows(:x86_64) => ("$bin_prefix/ImageMagick.v6.9.10-4.x86_64-w64-mingw32.tar.gz", "76bec9d440d8dd51395901e39405cac6e6c047f2a077a3590a96fe6847565fe2"),
+)
+
+# Install unsatisfied or updated dependencies:
+unsatisfied = any(!satisfied(p; verbose=verbose) for p in products)
+if haskey(download_info, platform_key())
+    url, tarball_hash = download_info[platform_key()]
+    if unsatisfied || !isinstalled(url, tarball_hash; prefix=prefix)
+        # Download and install binaries
+        install(url, tarball_hash; prefix=prefix, force=true, verbose=verbose)
+    end
+elseif unsatisfied
+    # If we don't have a BinaryProvider-compatible .tar.gz to download, complain.
+    # Alternatively, you could attempt to install from a separate provider,
+    # build from source or something even more ambitious here.
+    error("Your platform $(triplet(platform_key())) is not supported by this package!")
 end
 
-
-@BinDeps.install Dict([(:libwand, :libwand)])
-
-
-is_windows() && pop!(BinDeps.defaults)
+# Write out a deps.jl file that will contain mappings for our products
+write_deps_file(joinpath(@__DIR__, "deps.jl"), products)
```

Comment:
Will all the dependencies as well as ImageMagick itself try to create the same `deps.jl` file? Is that bad?

Reply:
I was wondering about this as well, but since it works and I'm not sure how to do it differently I was leaving it like that for now.. @staticfloat ?!

Reply:
Presumably you could have them each write their own `foo_deps.jl` file and then have one `deps.jl` that includes them all?

Reply:
Having multiple `deps.jl` files is only necessary if you want to manually `dlopen()` files from the other projects; right now you're only going to have access to variables that define the `libwand` variable; you won't get variables that point to the location of `zlib`, `libpng`, etc... I don't think you care about those, so this is fine, but this user experience is high on our list of things to improve.  :)